programming-extensions/programming-extension-dataspaces/src/main/java/org/objectweb/proactive/extensions/dataspaces/vfs/adapter/VFSFileObjectAdapter.java

```diff
@@ -597,33 +586,42 @@ public DataSpacesFileObject ensureExistingOrSwitch(boolean checkWriteProtected)
         return null;
     }
 
-    private static boolean checkIfFileWritable(DataSpacesFileObject dsfo) throws FileSystemException {
-        if (!dsfo.isWritable()) {
+    private static boolean checkIfFileWritable(DataSpacesFileObject fileObject) throws FileSystemException {
```

Comment:
The purpose of the method is to check permissions the goal is not to delegate the checking at higher level otherwise the same extra piece of code for performing the check will be duplicated at different places.
 
Regarding `DataSpacesFileObject` result, it cannot be trusted because for some systems (e.g. Windows) and network file systems it always return `true`. The same apply with the JDK method `File.canWrite`. That's why as a last try it is tried to create a true file.